clients/src/main/java/org/apache/kafka/clients/consumer/internals/Acknowledgements.java

```diff
@@ -115,25 +120,26 @@ public int size() {
      * @return Whether the acknowledgements were sent to the broker and a response received
      */
     public boolean isCompleted() {
-        return acknowledgeErrorCode != null;
+        return completed;
     }
 
     /**
-     * Set the acknowledgement error code when the response has been received from the broker.
+     * Set the acknowledgement exception when the response has been received from the broker.
      *
-     * @param acknowledgeErrorCode the error code
+     * @param acknowledgeException the exception (will be null if successful)
      */
-    public void setAcknowledgeErrorCode(Errors acknowledgeErrorCode) {
-        this.acknowledgeErrorCode = acknowledgeErrorCode;
+    public void setAcknowledgeException(KafkaException acknowledgeException) {
+        completed = true;
+        this.acknowledgeException = acknowledgeException;
     }
 
     /**
-     * Get the acknowledgement error code when the response has been received from the broker.
+     * Get the acknowledgement exception when the response has been received from the broker.
      *
      * @return the error code
      */
-    public Errors getAcknowledgeErrorCode() {
-        return acknowledgeErrorCode;
+    public KafkaException getAcknowledgeException() {
```

Comment:
We generally avoid `get` prefix in methods so should it be as like below?
```suggestion
    public KafkaException acknowledgeException() {
```

Reply:
In public classes, yes, I agree. However, here there are many accessor methods already present and I don't think renaming them all improves things.